packages/CodingStandard/src/Fixer/Property/ArrayPropertyDefaultValueFixer.php

```diff
@@ -0,0 +1,103 @@
+<?php declare(strict_types=1);
+
+namespace Symplify\CodingStandard\Fixer\Property;
+
+use Nette\Utils\Strings;
+use PhpCsFixer\DocBlock\DocBlock;
+use PhpCsFixer\Fixer\DefinedFixerInterface;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use SplFileInfo;
+
+final class ArrayPropertyDefaultValueFixer implements DefinedFixerInterface
+{
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Array property should have default value, to prevent undefined array issues.',
+            []
+        );
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        // analyze only properties with comments
+        return $tokens->isAllTokenKindsFound([T_DOC_COMMENT, T_VARIABLE]);
+    }
+
+    public function isRisky(): bool
+    {
+        return false;
+    }
+
+    public function fix(SplFileInfo $file, Tokens $tokens): void
+    {
+        foreach ($tokens as $index => $token) {
+            if (! $token->isGivenKind(T_DOC_COMMENT)) {
+                continue;
+            }
+
+            if (! $this->isArrayPropertyDocComment($token)) {
+                continue;
+            }
+
+            $semicolonSignOrArrayOpenerPosition = $tokens->getNextMeaningfulToken($index + 4);
+            $semicolonSignOrArrayOpenerToken = $tokens[$semicolonSignOrArrayOpenerPosition];
+            if ($semicolonSignOrArrayOpenerToken->equals('[')) {
```

Comment:
next bug.

```php
$foo = [];  // case 1
$foo = array(); // case 2
$foo = null; // case 3
$foo; // case 4
```

so what actually you wanna detect?
case 4.

what you are detecting? nothing. this condition is always false. as `[` is array index opening token.
as you saw previously, short array opening token is `CT::T_ARRAY_SQUARE_BRACE_OPEN`

instead of trying to guess that there is no need for fixing, check the opposite - is there a need for fix:
```php
if (! $tokenAfterVariable->equals(';')) {
    $this->insert...
}